[macos] Added Connectivity

AndreiMisiukevich opened the pull request:
### Description of Change ###

Added implementation of Connectivity plugin for macOs platforms

### Bugs Fixed ###

- Related to issue https://github.com/xamarin/Essentials/issues/111

### API Changes ###

None (just added macOs support)

### Behavioral Changes ###

Try to build **Xamarin.Essentials.Additional** and run **Mac** sample. Open **Connectivity** tab:
Actual behavior: crash
Expected: it shows info re network status

<img width="624" alt="2" src="https://user-images.githubusercontent.com/10124814/53607459-e3f0f800-3bcf-11e9-8141-09494714a81a.png">
<img width="558" alt="1" src="https://user-images.githubusercontent.com/10124814/53607460-e4898e80-3bcf-11e9-915a-97eb6a7b4b63.png">

knocte commented:
Did you take some code from https://github.com/jamesmontemagno/ConnectivityPlugin/tree/master/src/Connectivity.Plugin.Shared.Apple ? if yes, you should mention this in the commit message, to have proper author information

AndreiMisiukevich commented:
@knocte thank you.
I will update PR :)

AndreiMisiukevich committed:
[macos] Added Connectivity
# iOS code is adapted for macos (https://github.com/mattleibow/Essentials/tree/master/Xamarin.Essentials/Connectivity )

AndreiMisiukevich commented:
Hey
Is there any update?

knocte commented:
Seems this has conflicted with this: https://github.com/mattleibow/Essentials/commit/af18956e726b1ec6f77afb29300fc8bd84cde54f

AndreiMisiukevich commented:
it doesn't make sense anymore
